guides/advanced_user_guide.md

```diff
@@ -0,0 +1,112 @@
+
+### Basic Usage
+[The flip-coin pipeline using custom task](/samples/flip-coin-custom-task) example is a basic tutorial on how to use a pre-defined [CEL custom task](https://github.com/tektoncd/experimental/tree/main/cel) to replace the traditional KFP-Tekton conditions. Behind the scene, the pre-defined CEL custom task is written as the below containerOp in [/sdk/python/kfp_tekton/tekton.py](/sdk/python/kfp_tekton/tekton.py):
+```python
+def CEL_ConditionOp(condition_statement):
+    '''A containerOp template for CEL and converts it into Tekton custom task
+    during Tekton compiliation.
+    Args:
+        condition_statement: CEL expression statement using string and/or pipeline params.
+    '''
+    ConditionOp = dsl.ContainerOp(
+            name="condition-cel",
+            image=CEL_EVAL_IMAGE,
+            command=["sh", "-c"],
+            arguments=["--apiVersion", "cel.tekton.dev/v1alpha1",
+                       "--kind", "CEL",
+                       "--name", "cel_condition",
+                       "--%s" % DEFAULT_CONDITION_OUTPUT_KEYWORD, condition_statement],
+            file_outputs={DEFAULT_CONDITION_OUTPUT_KEYWORD: '/tmp/tekton'}
+        )
+    ConditionOp.add_pod_annotation("valid_container", "false")
+    return ConditionOp
+```
+
+Here, the `apiVersion`, `kind`, and `name` are mandatory fields for all custom tasks. The `DEFAULT_CONDITION_OUTPUT_KEYWORD` field is a required argument only for running CEL custom task. The `CEL_EVAL_IMAGE` image is registered as a custom task in the KFP-Tekton compiler, so the backend knows this ContianerOp needs to be compiled differently.
+
+### Defining Your Own Tekton Custom Task on KFP-Tekton
+1. To define your own Tekton custom task on KFP-Tekton, first define a unique image name for classifying your containerOp definition as a custom task and update the list of registered custom task images.
+    ```python
+    MY_CUSTOM_TASK_IMAGE_NAME = "veryunique/image:latest"
+    from kfp_tekton.tekton import TEKTON_CUSTOM_TASK_IMAGES
+    TEKTON_CUSTOM_TASK_IMAGES = TEKTON_CUSTOM_TASK_IMAGES.append(MY_CUSTOM_TASK_IMAGE_NAME)
+    ```
+2. Define your own Tekton custom task as a containerOp following the below constraints:
+    ```python
+    ConditionOp = dsl.ContainerOp(
+            name="any-name",
+            image=MY_CUSTOM_TASK_IMAGE_NAME,
+            command=["any", "command"],
+            arguments=["--apiVersion", "custom_task_api_version",
+                       "--kind", "custom_task_kind",
+                       "--name", "custom_task_name",
+                       "--taskSpec", {"task_spec_key": "task_spec_value"},
+                       "--other_custom_task_argument_keys", custom_task_argument_values],
+            file_outputs={"other_custom_task_argument_keys": '/anypath'}
+        )
+    # Annotation to tell the Argo controller that this containerOp is for specific Tekton runtime only.
```

Comment:
Argo?

Reply:
Sorry for misunderstanding, yes this is an additional annotation for Argo.

Reply:
yes, for Tekton you don't really need to add this annotations. We have this annotation in the example to showcase how we are maintaining the DSL support when compiling with the Argo backend.